### PR TITLE
Add semantic test detection for C#, Go, Rust, PHP, Python, Java, Scala via Tree-sitter

### PR DESCRIPTION
**Pull request by DavidBakerEffendi**
This PR introduces semantic test detection across several languages by leveraging Tree-sitter queries. Previously, identifying test files relied primarily on filename-based heuristics in `ContextManager`. This change adds a `containsTests` method to `IAnalyzer` and implements it for C#, Go, and Rust.

Key implementation details:
- **C#**: Detects `[Test]`, `[Fact]`, or `[Theory]` attributes on methods.
- **Go**: Identifies functions matching the `func TestXxx(t *testing.T)` signature using a combination of Tree-sitter captures and string-matching-based filtering to exclude benchmarks and methods with receivers.
- **Rust**: Detects `#[test]` and `#[cfg(test)]` attributes.
- **Java**: Detects `@Test`, `@ParameterizedTest`, and `@RepeatedTest` attributes
- **Python**: Detects methods with a `test_` suffix or PyTest marker.
- **PHP**: Detects Docblock/Comment-based `@test_marker` and if a function name contains `test`
- **Scala**: Detects same annotations as Java, but includes looking for `scalatest`-style markers and infix expressions such as `it` `should` etc.
- **Infrastructure**: Updates `TreeSitterAnalyzer` to track this state during the analysis pass and persists it in the `AnalyzerStateDto` for caching.
- **ContextManager**: Updated `isTestFile` to identify whether a file is a test by prioritizing semantic insights from an available analyzer and falling back to path-based regex heuristics if the analyzer is missing or yields no results.

### CODE REVIEW

**Comment by jbellis** [comment]
is there a reason not to add @Test detection for Java as well?

**Comment by DavidBakerEffendi** [comment]
(Converting to draft as I support more languages)

**Comment by DavidBakerEffendi** [comment]
Example of `Run Tests` enabled on some of my AoC 2025 Rust code
Screenshot 2026-01-05 at 11 44 48

**Comment by jbellis** [comment]
### What additional tests would add the most value?

You already added solid “happy path + integration” tests per language. The most valuable additions would be **edge-case / regression tests** where false positives are likely.

#### Go (highest value)
1. **Multiple parameter names sharing one type**:
   - `func TestX(a, b *testing.T) {}` should **not** be detected.
2. **Extra parameters**:
   - `func TestX(t *testing.T, x int) {}` should **not** be detected.
3. **Generic tests** (if Go grammar supports type parameters in your version):
   - `func TestX[T any](t *testing.T) {}` should still be detected (or explicitly not, depending on desired behavior).

#### Python
1. Decorators that are **pytest-related but not marks**:
   - `@pytest.fixture` should **not** mark as tests if your intended rule is specifically `pytest.mark.*`.
2. A function named `test` (without underscore):
   - Decide whether `def test():` should count; if not, add a negative test to lock it in.

#### PHP
1. `@test` comment not adjacent to a function/method should **not** mark the file as containing tests.
   - e.g., file header docblock contains “@test” as plain text.
2. Ensure adjacency logic works for typical PHP AST layout (docblock inside class bodies, etc.), especially if the grammar doesn’t create “whitespace” nodes (your loop tries to skip them).

#### Scala
1. Import containing `org.scalatest` in a non-test context still marks test (expected today).
   - If that’s intentional, add a test explicitly showing it.
2. Annotation `@Test` in a non-test annotation namespace vs JUnit import presence:
   - If you want to avoid false positives, you’d need more sophistication; otherwise, a test can document the current behavior.

**Comment by DavidBakerEffendi** [comment]
@jbellis Improved test coverage and made Go and Scala more precise

File: app/src/main/java/ai/brokk/ContextManager.java [hunk 1]
```diff
@@ -101,8 +101,21 @@ public class ContextManager implements IContextManager, AutoCloseable {
...
+        if (analyzer != null && !analyzer.isEmpty()) {
+            if (analyzer.containsTests(file)) {
+                return true;
+            }
+            // Analyzer exists but says no tests — could still fall through
+            // to heuristics for languages where analyzer doesn't implement this
+        }
 
+        // 2. Filename/path heuristics as fallback
         return TEST_FILE_PATTERN.matcher(file.toString()).matches();
```
Comment (cmanto) [inline app/src/main/java/ai/brokk/ContextManager.java]: This seems like it should be delegated to the analyzer? can a single regex be usable for a number of languages?

File: app/src/main/java/ai/brokk/analyzer/CSharpAnalyzer.java [hunk 2]
```diff
@@ -243,4 +247,20 @@ protected String formatFieldSignature(
     public Optional<String> extractCallReceiver(String reference) {
         return ClassNameExtractor.extractForCSharp(reference);
     }
+
+    @Override
+    protected boolean containsTestMarkers(TSTree tree) {
+        TSQuery query = getThreadLocalQuery();
+        TSQueryCursor cursor = new TSQueryCursor();
```
[the remark this answers is not in the document]
Comment (DavidBakerEffendi) [reply app/src/main/java/ai/brokk/analyzer/CSharpAnalyzer.java]: Ah yes re-discovered the reason. So at first I saw TSA wasn't cleaning its cursor and then looked in the source code. Found that it registers clean-up with dereferencing/garbage collection.
```
    private TSQueryCursor(long ptr, long progressPayloadPtr) {
        this.ptr = ptr;
        this.progressPayloadPtr = progressPayloadPtr;
        CleanerRunner.register(this, new TSQueryCursorCleanAction(ptr, progressPayloadPtr));
    }
```
I think review LLMs maybe confuse this with the usual database query cursor terminology which do have explicit close protocols.
